client/asset/btc/btc.go

```diff
@@ -363,10 +363,20 @@ type block struct {
 // which is added to the findRedemptionQueue with the contract outpoint as
 // key.
 type findRedemptionReq struct {
+	ctx          context.Context
 	contractHash []byte
 	resultChan   chan *findRedemptionResult
 }
 
+func (frr *findRedemptionReq) contextCancelled() bool {
+	select {
+	case <-frr.ctx.Done():
+		return true
+	default:
+		return false
+	}
```

Comment:
Agreed.  I'd even go so far to say you could get rid of the function since it's used in just one place.  But it's fine to hide the context details.  To that extent though, just `canceled()` would be more appropriate.

Reply:
My brain must have been clogged: I had to write a select to know if a context has been canceled! 😆